### PR TITLE
Docker multistage build, decr image size, add docker-compose, add CI

### PR DESCRIPTION
**Pull request by pratikbalar**
- Add multistage build
- Tried to decrease image size around `122MB`
- Add tini as process initializer <https://github.com/krallin/tini#why-tini>
- Add `docker-compose.yaml`
- Add CI to test container image build, run and test it with curl

### CODE REVIEW

**Comment by pratikbalar** [comment]
@reiz

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  nginx_proxy:
+    image: nginx_proxy:latest
+    build:
+      context: .
+      dockerfile: Dockerfile
+    container_name: nginx_proxy
+    volumes:
+      - ./nginx_whitelist.conf:/usr/local/nginx/conf/nginx.conf
+    environment:
+      - TZ=Asia/Kolkata
```
Comment (reiz) [inline docker-compose.yaml:12]: I guess this is your time zone. Could you please change that to `Etc/UTC`. That would be more universal.

**Review by reiz — CHANGES_REQUESTED** [review]
Looks pretty good. Just a little change is requested.

File: docker-compose.yaml (hunk 1 above)
Comment (pratikbalar) [reply docker-compose.yaml]: Done

**Comment by pratikbalar** [comment]
hey @reiz, do you want to add docker/container image pushing to registry in CI ?

**Comment by reiz** [comment]
> * Do you want to add docker/container image pushing to registry in CI ?

That would be interesting. But not necessarily on the master branch. Would create a dev branch for that and then tag the Docker images with the commit hash.

**Comment by reiz** [comment]
> * Interested in Alpine based image?

Yes. Alpine images would be definitively interesting!

**Comment by reiz** [comment]
@pratikbalar Congrats! First build on GitHub Action was successful! I will build a new Docker image next week & publish it to Docker Registry. For the next days I'm on vacation.

**Comment by pratikbalar** [comment]
> That would be interesting. But not necessarily on the master branch. Would create a dev branch for that and then tag the Docker images with the commit hash.

then I think add master branch image push manually with CI, so there will be human gate

**Comment by reiz** [comment]
@pratikbalar I just merged the last PR and build & published a docker image to Docker Hub. The README.md is updated.

**Comment by pratikbalar** [comment]
@reiz 🎉

**Pull request merged**